bots/pr/src/main/java/org/openjdk/skara/bots/pr/CommandExtractor.java

```diff
@@ -52,7 +52,10 @@ static List<CommandInvocation> extractCommands(Map<String, CommandHandler> comma
                     multiLineHandler = null;
                 }
                 var command = commandMatcher.group(1).toLowerCase();
-                var handler = commandHandlers.get(command);
+                var handler = commitCommandFirst ? CommitCommandWorkItem.commandHandlers.get(command) : PullRequestCommandWorkItem.commandHandlers.get(command);
```

Comment:
What are you trying to achieve with the commitCommandFirst boolean? In what situation would it be wrong to just use one unified list?

Reply:
Hm, now I realize what you mean about the HelpCommand. If you invoke `/help` on a commit now, you get all the PR commands listed. That's not good. We still need some way of separating which commands should be listed for help with a commit and help with a PR.

Reply:
> The only one situation I concern about is the help command. But after second thought, it looks good. So I unified them.

In detail, the code logic of the master branch:

1. use `help` command on PR which has **not** been integrated

the bot prints the commands which are **allowed in PR** and **the extranal PR command**

2. use `help` command on PR which **has been integrated**

the bot prints the commands which are **allowed in commit** and **the extranal PR command**

3. use `help` command on a commit

the bot prints the commands which are **allowed in commit** and **the extranal commit command**

But the logic of this patch is:
1. use `help` command on PR which has **not** been integrated (unchanged)

the bot prints the commands which are **allowed in PR** and **the extranal PR command**

2. use `help` command on PR which **has been integrated** or use `help` **command on a commit**

the bot prints the commands which are **allowed in commit** and **the extranal commit command**

So there is a little diffrence between the master branch and my patch. But the logic in my patch seems good.

> Hm, now I realize what you mean about the HelpCommand. If you invoke `/help` on a commit now, you get all the PR commands listed. That's not good. We still need some way of separating which commands should be listed for help with a commit and help with a PR.

From your description, I think one among us misunderstood the code. Did I miss something?

Reply:
You are right, I missed that you added filtering on allowedIn*. That was an elegant solution, thanks!